### DISCUSSION TITLE
Testing Bonsai workflow/Harp

### DISCUSSION THREAD

**Discussion by FannyCazettes** [harp-tech/protocol#82] [Q&A]
We have implemented a new task with Harp and Bonsai that seems to function well but now we would like to run some sanity tests to make sure there are no weird bugs (like delays in water delivery, bugs when writing the rotary encoder signal, control of the motor, etc.).  Are there some Bonsai functions that simulate the board's signals during the task to allow testing a large number of trials without having to run the task by hand? Otherwise, what is the best approach to test the setup? 
Thanks.

**Comment by bruno-f-cruz** [comment]
Heya,

Not sure I understand the full scope of the question. I guess you could use the "format" and "createmessage" nodes to create harp messages. Since the devices also communicate using these data types you can "fake" any type of data you want. Regarding benchmarks, I guess it depends what you want to see, if you want to check water delivery, I would probably use an oscilloscope to look at the output signal. The rotary encoder, you could save it using an independent Nidaq/oscilloscope and reconstitute the pulses -> position offline.
